header_test.go

```diff
@@ -41,6 +50,18 @@ func TestReadTimeoutV1Invalid(t *testing.T) {
 	}
 }
 
+func TestReadTimeoutPropagatesReadError(t *testing.T) {
+	var e errorReader
+	reader := bufio.NewReader(&e)
+	_, err := ReadTimeout(reader, 50*time.Millisecond)
+
+	if err == nil {
+		t.Fatalf("expected error %s", errReadIntentionallyBroken)
+	} else if err != errReadIntentionallyBroken {
+		t.Fatalf("expected error %s, actual %s", errReadIntentionallyBroken, err)
+	}
+}
```

Comment:
This is OK but I'd be happier w/ a test case that covers the real scenario you ran into, the TLS issue.